### PR TITLE
fix: api routes not found after apply intl middleware

### LINKED ISSUE #209: Should Intl middleware be applied to API routes?
# Problem

Glad to see you added i18n to this boilerplate.

I noticed in `middleware.ts` that `intlMiddleware` in `beforeAuth` will be applied on every request, even if it is an API route.

https://github.com/ixartz/Next-js-Boilerplate/blob/04fba59b5a4061e015a8796412dbfd3bbaeeeecf/src/middleware.ts#L13-L23

Since the API route does not contain the `locale` parameter, this causes a 404 to be returned when using `intlMiddleware`.

The example API route `/api/guestbook` is placed in the ignore list, therefore does not apply any middleware, perhaps causing you to overlook this issue.

# Reproduce

Removing `/api/guestbook` from `ignoredRoutes` should reproduce the issue. (which mark it as public route)

# Potential Fix

I simply added a condition to it to only execute in non-API routes, not sure if there is a better solution.

```ts
beforeAuth: (req) => {
    // Execute next-intl middleware before Clerk's auth middleware
    if (!req.nextUrl.pathname.includes('/api')) {
      return intlMiddleware(req);
    }
    return undefined;
  },

### PR DESCRIPTION
**Pull request by sunxyw**
Move API Routes into `[locale]`

fix #209

### CODE REVIEW

**Comment by sunxyw** [comment]
I simply moved the API routes to `[locale]` and removed `/api/guestbook` from `ignoredRoutes` in `middleware.ts`.

I'm not sure if I should add a new Route Handler to demonstrate the use of i18n.

**Comment by ixartz** [comment]
The PR looks good to me, thank you so much for creating the issue and for your help. I'll merge it when I go back to my computer.

For now, no need to add a new Route Handler to demonstrate the use of i18n.

File: src/middleware.ts [hunk 1]
```diff
@@ -13,7 +13,6 @@ const intlMiddleware = createMiddleware({
 export default authMiddleware({
   publicRoutes: (req: NextRequest) =>
     !req.nextUrl.pathname.includes('/dashboard'),
-  ignoredRoutes: ['/api/guestbook'],
   // By default, the middleware will return a 401 response for all routes `/api/*` when the user is signed out.
```
Comment (ixartz) [inline src/middleware.ts:16]: Do you think you can also remove the comment below? it is not relevant anymore.

Thanks in advance.

Comment (ixartz) [reply src/middleware.ts:16]: `// By default, the middleware will return a 401 response for all routes `/api/*` when the user is signed out.
// But, for `/api/guestbook`, we want unauthenticated users to be able to access it.`

Comment (sunxyw) [reply src/middleware.ts]: Removed.

**Comment by ixartz** [comment]
@sunxyw Thank you so much for reporting and solving the issue.

**Pull request merged**